### DISCUSSION TITLE
How to know if a procedure is a query or mutation clientside

### DISCUSSION THREAD

**Discussion by kehwar** [trpc/trpc#5104] [Q&A]
I want utility to call my procedures without knowing if it the procedure is a query or a mutation, since the client procedure is a proxy I cannot check by property

The best I could do was to try each function inside a try catch

**Comment by KATT** [comment]
The trpc client has no runtime information about the server, so this isn't possible without you doing a script or something that spits this out somewhere.

What is the problem you're trying to solve with this?

**Comment by kehwar** [comment]
I have a flat router with a few queries and mutations

Instead of this `client.getUser.query(args)` and `client.updateUser.mutate(args)`

I want this  `client.getUser(args)` and `client.updateUser(args)`

I don't care about losing the options

**Reply by Nick-Lucas** [reply]
To expand on what Katt said. tRPC uses proxies for the client code, this means the functions you’re calling aren’t real, they’re dynamically calculated. Queries and mutations are GET and POST calls, and so the name of the method you call informs the proxy which verb to use (among other things) 

So in short this will probably never change in tRPC

**Reply by KATT** [reply]
Have a look at #2463 for inspiration on how you could do this

**Reply by kehwar** [reply]
Thanks, I didn't found that thread in my serach

**Comment by kehwar** [comment]
> Have a look at #2463 for inspiration on how you could do this
